promql/testdata/legacy.test

```diff
@@ -163,6 +163,16 @@ eval instant at 50m delta(http_requests[25m], 1)
 	{group="production", instance="1", job="api-server"} 100 
 	{group="production", instance="1", job="app-server"} 300 
 
+eval instant at 50m area(http_requests[25m])
```

Comment:
Could you create a new file (`functions.test` or something alike)?
This one is called `legacy.test` because its a direct port of the old tests that all operate on the same few inputs.
The new format allows us to write tests better targeted towards what's actually being tested.

Eventually those tests should be revisited and potentially rewritten with their own input into a new file. Thus, I'd like to avoid growing it.

Currently the `at 50m` is still non-optional. If it's manually set to the end of the input, we can remove it in those cases later.